test/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BusinessEntityControllerSpec.scala

```diff
@@ -746,9 +773,29 @@ class BusinessEntityControllerSpec
 
     }
   }
+  abstract class AddBusinessPageSetup(
```

Comment:
im happy to move the setup class further up in the test class, as you asked. One of the main issues i found was that we do not have one controller per page. This particular controller supports 4 pages so the test class ends up being huge and you end up with this issue. If we had one controller per page, this would be less of an issue because i prefer these setup classes at the bottom of the class.

Reply:
addressed